nipoppy/workflows/pipeline_store/zenodo.py

```diff
@@ -100,6 +102,36 @@ def run_main(self):
             )
             raise SystemExit(1)
 
+        if self.record_id:
+            self.record_id = self.record_id.removeprefix("zenodo.")
+            current_metadata = self.zenodo_api.get_record_metadata(self.record_id)
+            if not self.force and not _is_same_pipeline(
+                pipeline_config, current_metadata
+            ):
+                raise ZenodoAPIError(
+                    "The pipeline metadata does not match the existing record "
+                    f"(zenodo.{self.record_id}). Aborting."
+                    "\nUse the --force flag to force the update."
+                )
+        else:
+            pipeline_type = pipeline_config.PIPELINE_TYPE.value
+            pipeline_name = pipeline_config.NAME
+            pipeline_version = pipeline_config.VERSION
+            records = self.zenodo_api.search_records(
+                "",
+                keywords=[
+                    f"pipeline_type:{pipeline_type}",
+                    f"pipeline_name:{pipeline_name}",
+                    f"pipeline_version:{pipeline_version}",
+                ],
+            )["hits"]
+            if not self.force and len(records) > 0:
+                raise ZenodoAPIError(
+                    "It looks like this pipeline already exist in Zenodo. Aborting."
```

Comment:
It would be very nice if we could include the Zenodo ID(s) for the existing record(s) in the error message, so that if there is only one existing Zenodo ID then the user can then just copy-paste it the next time they run the command, instead of looking it up on Zenodo.

This would be just the IDs so we shouldn't get any weird parsing errors (hopefully).

Also, typo:
```suggestion
                    "It looks like this pipeline already exists in Zenodo. Aborting."
```